File: components/jpa-handling-session/src/main/java/com/sv/youapp/component/authorization/entities/jpa/UserEntity.java
package com.sv.youapp.component.authorization.entities.jpa;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.Set;

@Entity
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
@Table(name = "users", indexes = { @Index(columnList = "username", unique = true) })
public class UserEntity {

	@Id @GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

	@Column(nullable = false, unique = true, length = 100)
	private String username;

	@Column(nullable = false, length = 200)
	private String password;

	@Column(name = "profile_picture_url", length = 512)
	private String profilePictureUrl;

	@Column(nullable = false)
	private Boolean enabled;

	@Column(name = "registered_at", nullable = false)
	private Instant registeredAt;

	@ManyToMany
	@JoinTable(
		name = "user_roles",
		joinColumns = @JoinColumn(name = "user_id"),
		inverseJoinColumns = @JoinColumn(name = "role_id")
	)
	private Set<RoleEntity> roles;

	@ManyToMany
	@JoinTable(
		name = "user_authorities",
		joinColumns = @JoinColumn(name = "user_id"),
		inverseJoinColumns = @JoinColumn(name = "authority_id")
	)
	private Set<AuthorityEntity> authorities;
}

File: platform-instances/authorization-server/src/main/kotlin/com/sv/youapp/service/authorization/entities/jpa/UserEntity.kt
package com.sv.youapp.service.authorization.entities.jpa

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Index
import jakarta.persistence.JoinColumn
import jakarta.persistence.JoinTable
import jakarta.persistence.ManyToMany
import jakarta.persistence.Table
import java.time.Instant

@Entity
@Table(
    name = "users",
    indexes = [Index(columnList = "username", unique = true)],
)
class UserEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Integer,
    @Column(nullable = false, unique = true, length = 100)
    var username: String,
    @Column(nullable = false, length = 255)
    var password: String,
    @Column(nullable = false, unique = true, length = 255)
    var email: String,
    @Column(name = "profile_picture_url", length = 512)
    var profilePictureUrl: String?,
    @Column(nullable = false)
    var enabled: Boolean,
    @Column(name = "registered_at", nullable = false)
    var registeredAt: Instant,
    @ManyToMany
    @JoinTable(
        name = "user_roles",
        joinColumns = [JoinColumn(name = "user_id")],
        inverseJoinColumns = [JoinColumn(name = "role_id")],
    )
    var roles: MutableSet<RoleEntity>,
    @ManyToMany
    @JoinTable(
        name = "user_authorities",
        joinColumns = [JoinColumn(name = "user_id")],
        inverseJoinColumns = [JoinColumn(name = "authority_id")],
    )
    var authorities: MutableSet<AuthorityEntity>,
)
